src/x509.c

```diff
@@ -4765,70 +4769,100 @@ WOLFSSL_X509_NAME* wolfSSL_X509_get_subject_name(WOLFSSL_X509* cert)
     return NULL;
 }
 
+#if defined(OPENSSL_EXTRA) && defined(NO_SHA) && !defined(NO_SHA256)
+static unsigned long wolfSSL_X509_name_sha256hash(WOLFSSL_X509_NAME* name)
+{
+    unsigned long  hash = 0;
+    unsigned char* canonName = NULL;
+    byte           digest[WC_MAX_DIGEST_SIZE];
+    int            size = 0;
+
+    WOLFSSL_ENTER("wolfSSL_X509_name_sha256hash");
+
+    if (name == NULL) {
+        WOLFSSL_ERROR_MSG("WOLFSSL_X509_NAME pointer was NULL");
+        return 0;
+    }
+
+    if (name->sz == 0) {
+        WOLFSSL_ERROR_MSG("nothing to hash in WOLFSSL_X509_NAME");
+        return 0;
+    }
+
+    size = wolfSSL_i2d_X509_NAME_canon(name, &canonName);
+
+    if (size <= 0 || canonName == NULL){
+        WOLFSSL_ERROR_MSG("wolfSSL_i2d_X509_NAME_canon error");
+        return 0;
+    }
+
+    if (wc_Sha256Hash((const byte*)canonName, (word32)size, digest) == 0) {
```

Comment:
Is the only difference between this function and wolfSSL_X509_NAME_hash the use of wc_Sha256Hash vs. wc_ShaHash? If so, I would just factor this out into a single function with an additional parameter indicating which hash function to use. We can still do better re: repeating ourselves. You could do the same regarding issuer and subject name. Have another parameter that tells you which name to grab. There's less savings to be had there, though. wolfSSL_X509_(subject|issuer)_name_hash are already pretty small.

Reply:
The two functions are guarded by different if defs and would be messy to combine.

wolfSSL_X509_NAME_hash is guarded by ifdef OPENSSL_EXTRA

wolfSSL_X509_(subject|issuer)_name_hash are guarded by if defined(OPENSSL_EXTRA) && (!defined(NO_SHA) || !defined(NO_SHA256))